lang/java/avro/src/main/java/org/apache/avro/file/DataFileReader.java

```diff
@@ -170,7 +171,7 @@ public void seek(long position) throws IOException {
     vin = DecoderFactory.get().binaryDecoder(this.sin, vin);
     datumIn = null;
     blockRemaining = 0;
-    blockStart = position;
+    blockFinished();
```

Comment:
This has the same result as before the change, but I'm using `blockFinished()` for consistency.

Reply:
Looking at the call:

https://github.com/apache/avro/blob/6863074f20a7c6e4a34780877206723a5d3c4e24/lang/java/avro/src/main/java/org/apache/avro/file/DataFileReader.java#L243-L246

It can be that this results in calls to the filesystem. Was this causing an issue?

Reply:
No. Both the `tell()` and `available()` commands check what's currently in memory.

What's more, after the `seek()` call, `vin` is reinitialised. This causes the method `available()` to always return `0`.

As `tell()` returns the position given to `seek()` just before, the call to `blockFinished()` yields the same result as the original code.